serverless.yaml

```diff
@@ -299,10 +322,17 @@ resources:
           Type: COGNITO_USER_POOLS
           Name: Authorizer
           ProviderARNs:
-            - !Join [
-                '',
-                [!Sub 'arn:${AWS::Partition}:cognito-idp:${AWS::Region}:${AWS::AccountId}:userpool/', !Ref UserPool],
-              ]
+            - !If 
+              - isUsingDefaultUserPool
+              - !Join [
+                  '',
+                  [!Sub 'arn:${AWS::Partition}:cognito-idp:${AWS::Region}:${AWS::AccountId}:userpool/', !Ref UserPool],
+                ]
+              - !Join [
+                  '',
+                  [!Sub 'arn:${AWS::Partition}:cognito-idp:${AWS::Region}:${AWS::AccountId}:userpool/', !Ref ExtUserPoolId],
+                ]
+

```

Comment:
Can the `!If` be where `!Ref UserPool` is? That would prevent a bit of code duplication.